core/src/main/java/org/axonframework/eventhandling/saga/SagaLifecycle.java

```diff
@@ -164,4 +161,23 @@ protected void execute(Runnable task) {
             throw new SagaExecutionException("Exception while executing a task for a saga", e);
         }
     }
+
+    /**
+     * Retrieve a {@link String} denoting the type of this Aggregate.
+     *
+     * @return a {@link String} denoting the type of this Aggregate
+     */
+    protected abstract String type();
+
+    /**
+     * Retrieve a {@link Object} denoting the identifier of this Aggregate.
+     *
+     * @return a {@link Object} denoting the identifier of this Aggregate
+     */
+    protected abstract Object identifier();
```

Comment:
Suggestion: call this method `getSagaIdentifier`, so you won't have to have two implementations of the same behavior in `AnnotatedSaga`.

Reply:
Got one! 
And, I also see that the javadoc states this provides an Aggregate type/id i.o. that of a Saga. Changed that as well.